Bridge/src/protocol.cpp

```diff
@@ -0,0 +1,162 @@
+#include "protocol.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <iterator>
+
+#include "debug.hpp"
+
+void RxProtocol::reset()
+{
+    this->body_.clear();
+    this->next_part_count_ = 0;
+}
+
+std::optional<mqtt::message_ptr> RxProtocol::receive_packet(const Frame packet)
+{
+    if (packet.frame_type != FrameType::Message)
+        // Not implemented
+        return { };
+
+    if (!this->next_frame_count_)
```

Comment:
Comment above this to state what this group of if statements does would be good.